src/wpool/mongoose_wpool_rdbms.erl

```diff
@@ -0,0 +1,54 @@
+-module(mongoose_wpool_rdbms).
+-behaviour(mongoose_wpool).
+
+-export([init/0]).
+-export([start/4]).
+-export([stop/2]).
+
+init() ->
+    case ets:info(prepared_statements) of
+        undefined ->
+            Heir = case whereis(ejabberd_sup) of
+                       undefined -> [];
+                       Pid -> [{heir, Pid, undefined}]
+                   end,
+            ets:new(prepared_statements,
+                    [named_table, public, {read_concurrency, true} | Heir]),
+            ok;
+        _ ->
+            ok
+    end.
+
+start(Host, Tag, WpoolOpts, RdbmsOpts) ->
+    try do_start(Host, Tag, WpoolOpts, RdbmsOpts)
+    catch
+        Err -> {error, Err}
+    end.
+
+do_start(Host, Tag, WpoolOpts0, RdbmsOpts) ->
+    Backend =
+        case lists:keyfind(server, 1, RdbmsOpts) of
+            {_, ConnStr} when is_list(ConnStr) -> odbc;
+            {_, Tuple} when is_tuple(Tuple) -> element(1, Tuple)
+        end,
+
+    try mongoose_rdbms_backend:backend_name() of
+        Backend -> ok;
+        OtherBackend ->
+            throw(#{reason => "Cannot start an RDBMS connection pool: only one RDBMS backend can be used",
+                    opts => RdbmsOpts, new_backend => Backend, existing_backend => OtherBackend})
+    catch
+        error:undef ->
+            backend_module:create(mongoose_rdbms, Backend, [query, execute])
+    end,
+
+    Worker = {mongoose_rdbms, RdbmsOpts},
+    WpoolOpts = [{worker, Worker}, {pool_sup_shutdown, infinity} | WpoolOpts0],
+    Name = mongoose_wpool:make_pool_name(rdbms, Host, Tag),
+    case wpool:start_sup_pool(Name, WpoolOpts) of
+        {ok, Pid} -> {ok, {Pid, [{call_timeout, 60000}]}};
```

Comment:
Do we really want to hard code the `call_timeout` or is it only a default value?

Reply:
Ping? Hardcoded 60s sounds like a bad idea to me.

Reply:
It's a default value. I'll refactor it so that it's clearer and also addresses https://github.com/esl/MongooseIM/pull/2084#discussion_r220206720